pos_stock_available_online/models/stock_quant.py

```diff
@@ -0,0 +1,40 @@
+import logging
+
+from odoo import models
+
+_logger = logging.getLogger(__name__)
+
+
+class StockQuant(models.Model):
+    _inherit = "stock.quant"
+
+    def _prepare_pos_message(self):
+        """
+        Return prepared message to send to POS
+        """
+        self.ensure_one()
+        return self.warehouse_id._prepare_vals_for_pos(self.product_id)
+
+    def _notify_pos(self):
+        """
+        Send notification to POS
+        """
+        pos_session_obj = self.env["pos.session"]
+        for quant in self:
+            warehouse_id = quant.warehouse_id.id
+            configs = pos_session_obj.search(
+                [
+                    ("state", "=", "opened"),
+                    ("config_id.display_product_quantity", "=", True),
+                    "|",
+                    ("config_id.additional_warehouse_ids", "in", [warehouse_id]),
+                    ("config_id.main_warehouse_id", "=", warehouse_id),
+                ],
+            ).mapped("config_id")
+            if configs:
```

Comment:
you got a for loop inside the _notify_available_quantity, if not record on the set, then will not perform any 
```suggestion
            configs._notify_available_quantity(quant._prepare_pos_message())
```

Reply:
this check is added to prevent the `_prepare_pos_message` method for the current quant from triggering unnecessarily if no config is found.